linode/step_create_image.go

```diff
@@ -17,25 +17,30 @@ func (s *stepCreateImage) Run(ctx context.Context, state multistep.StateBag) mul
 	c := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
 	disk := state.Get("disk").(*linodego.InstanceDisk)
+	instance := state.Get("instance").(*linodego.Instance)
 
 	ui.Say("Creating image...")
 	image, err := s.client.CreateImage(ctx, linodego.ImageCreateOptions{
 		DiskID:      disk.ID,
 		Label:       c.Label,
 		Description: c.Description,
 	})
+
+	if err == nil {
+		_, err = s.client.WaitForInstanceDiskStatus(ctx, instance.ID, disk.ID, linodego.DiskReady, 600)
+	}
+
+	if err == nil {
+		image, err = s.client.GetImage(ctx, image.ID)
```

Comment:
This logic was borrowed from the terraform provider.

After polling for the disk to become ready again, it is necessary to fetch the image again to make sure that image succeeded.  If GetImage 404's, then the image was revoked due to some processing error (too large after compression, typically).